docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+
+### grpc.block_results_service.enabled
+The gRPC block results service returns block results for a given height. If no height is given, it will return the block
+results from the latest height.
+```toml
+enabled = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+If [`grpc.laddr`](#grpcladdr) is empty, this setting is ignored and the service is not enabled.
+
+### grpc.privileged.laddr
+Configuration for privileged gRPC endpoints, which should **never** be exposed to the public internet.
+```toml
+laddr = ""
+```
+
+| Value type          | string                                                  |
+|:--------------------|:--------------------------------------------------------|
+| **Possible values** | TCP Stream socket (e.g. `"tcp://127.0.0.1:26662"`)      |
+|                     | Unix domain socket (e.g. `"unix:///var/run/abci.sock"`) |
+|                     | `""`                                                    |
+
+If not specified, the gRPC privileged endpoints will be disabled.
+
+### grpc.privileged.pruning_service
+Configuration specifically for the gRPC pruning service, which is considered a privileged service.
+```toml
+enabled = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+Only controls whether the pruning service is accessible via the gRPC API - not whether a previously set pruning service
+retain height is honored by the node. See the [storage.pruning](#storagepruninginterval) section for control over pruning.
+
+If [`grpc.laddr`](#grpcladdr) is empty, this setting is ignored and the service is not enabled.
+
+## Peer-to-peer
+
+These configuration options change the behaviour of the peer-to-peer protocol.
+
+### p2p.laddr
+
+TCP socket address for the P2P service to listen on and accept connections.
+```toml
+laddr = "tcp://0.0.0.0:26656"
+```
+
+| Value type          | string                                            |
+|:--------------------|:--------------------------------------------------|
+| **Possible values** | TCP Stream socket (e.g. `"tcp://127.0.0.1:26657"`)     |
+
+### p2p.external_address
+
+
+TCP address that peers should use in order to connect to the node.
+This is the address that the node advertises to peers.
+If not set, the [`p2p.laddr`](#p2pladdr) is advertised.
+
+Useful when the node is running on a non-routable address or when the
+node does not have the capabilities to figure out its IP public address.
+For example, this is useful when running from a cloud service (e.g, AWS, Digital Ocean).
+In these scenarios, the public or external address of the node should be set to
+`p2p.external_address`, while the private or internal address should be used as
+the listen address ([`p2p.laddr`](#p2pladdr)).
```

Comment:
Unless you're using some NAT hole poking (which is not mentioned in this text), if you bind your node to the internal or private address, remote node won't be able to reach you.
Probably what the test should rather say is that `p2p.laddr` should be bound to the `0.0.0.0` address (i.e., INADDR_ANY), as is the default BTW.

Reply:
The question here is not the listening address, but the authentication address.